### PR TITLE
Support X-Forwarded-For & more configurable rate limits

### PR DESCRIPTION
**Pull request by stephank**

### CODE REVIEW

File: config.toml.dist [hunk 1]
```diff
@@ -173,6 +180,73 @@ session_ttl = 900 # 15 minutes
 # Minimum cache time for downstream HTTP requests made by the broker
 cache_ttl = 3600 # 1 hour
 
+################################################################
+# Rate limits
+
+# Limits can be set in a flexible way. We've tried to provide some sensible
+# defaults, but if you're exposing your broker to the internet, you may need
+# control over limits when the need arises. This section details the available
```
Comment (onli) [inline config.toml.dist:188]: I will mention the minor and subjective stuff. 

_when the need arises_ could be left out.

File: config.toml.dist [hunk 2]
```diff
@@ -173,6 +180,73 @@ session_ttl = 900 # 15 minutes
...
+# options for rate limiting.
```
Comment (onli) [inline config.toml.dist:189]: _This section details the available  options for rate limiting._ could also be left out, the headline took care of that.

File: config.toml.dist [hunk 3]
```diff
@@ -173,6 +180,73 @@ session_ttl = 900 # 15 minutes
...
+# - `d` / `day` / `days`
+#
+# Flags control what the limit applies to and its behavior. The following flags
+# can be used:
+#
+# - `ip`: Apply the limit to the users IP address.
+# - `email`: Apply the limit to the users email address.
+# - `domain`: Apply the limit to the users email domain.
+# - `origin`: Apply the limit to the Relying Party origin.
+# - `decr_complete`: Decrement the counter for completed requests.
```
Comment (onli) [inline config.toml.dist:211]: I stumbled here. But with the example the description is correct.

File: config.toml.dist [hunk 4]
```diff
@@ -173,6 +180,73 @@ session_ttl = 900 # 15 minutes
...
+# - `extend_window`: Extend the window on every hit, instead of just the first.
```
Comment (onli) [inline config.toml.dist:212]: Same here. The examples are really important :)

**Comment by stephank** [comment]
I tried to simply reorder things, start with the examples, then the details. Hopefully that makes things a bit better?

**Comment by onli** [comment]
I think it's completely okay.

**Pull request merged**